src/cider/nrepl/middleware/debug.clj

```diff
@@ -438,12 +438,28 @@
             (eval form)
             (alter-meta! v assoc ::instrumented instrumented)))))))
 
+(def step-in-blacklist
+  "A set of names or regexes specifying namespaces that should not be stepped in
+  to. By default, this contains only \"clojure.core\"."
+  (atom #{"clojure.core"}))
+
+(defn blacklisted-for-step-in?
```

Comment:
No, the crash can definitely happen when you manually instrument. But to me, the difference is that it's very easy to step in to a core function by accident. There are probably many functions in clojure.core that are perfectly safe to debug, if someone really wants do explicitly do so, so I'm not sure we should make that impossible.